test/periodic.jl

```diff
@@ -2,47 +2,32 @@
     @testset "_periodic" begin
         @testset "$f" for f in (sin, cos)
             # A typical 24-hour day
-            for h in 0:24
-                result = _periodic(
-                    f,
-                    ZonedDateTime(2015, 6, 1, tz"America/Winnipeg") + Hour(h),
-                    Day(1),
-                )
-                expected = f(2π / 24 * h)
-                @test result ≈ expected atol=1e-14
-            end
+            x = ZonedDateTime(2015, 6, 1, tz"America/Winnipeg") .+ collect(Hour.(0:2))
+            result = _periodic.(f, x, Day(1))
+            expected = Dict(sin => [0., 0.2588, 0.5], cos => [1.0, 0.9659, 0.866])[f]
+            @test result ≈ expected atol=1e-4
 
             # A spring daylight-saving time change where the day has 23-hours
-            for h in 0:23
-                result = _periodic(
-                    f,
-                    ZonedDateTime(2015, 3, 8, tz"America/Winnipeg") + Hour(h),
-                    Day(1),
-                )
-                expected = f(2π / 23 * h)
-                @test result ≈ expected atol=1e-14
-            end
+            x = ZonedDateTime(2015, 3, 8, tz"America/Winnipeg") .+ collect(Hour.(0:2))
+            result = _periodic.(f, x, Day(1))
+            expected = Dict(sin => [0., 0.2698, 0.5196], cos => [1., 0.9629, 0.8544])[f]
+            @test result ≈ expected atol=1e-4
 
             # A fall daylight-saving time change where the day has 25-hours
-            for h in 0:25
-                result = _periodic(
-                    f,
-                    ZonedDateTime(2015, 11, 1, tz"America/Winnipeg") + Hour(h),
-                    Day(1),
-                )
-                expected = f(2π / 25 * h)
-                @test result ≈ expected atol=1e-14
-            end
+            x = ZonedDateTime(2015, 11, 1, tz"America/Winnipeg") .+ collect(Hour.(0:2))
+            result = _periodic.(f, x, Day(1))
+            expected = Dict(sin => [0., 0.2487, 0.4818], cos => [1., 0.9686, 0.8763])[f]
+            @test result ≈ expected atol=1e-4
         end
 
         @testset "phase shift" begin
-            @test _periodic(sin, DateTime(1), Week(1)) == 0
-            @test _periodic(sin, DateTime(1), Week(1), Day(7)) == 0
-            @test _periodic(sin, DateTime(1), Week(1), Day(-7)) == 0
+            @test _periodic(sin, DateTime(2018), Week(1)) == 0
+            @test _periodic(sin, DateTime(2018), Week(1)) == 0
+            @test _periodic(sin, DateTime(2018), Week(1)) == 0
```

Comment:
are these supposed to take `Day(7)`, `Day(-7)`?

Reply:
Oops. Copy-paste fail.